irohad/consensus/yac/transport/impl/network_impl.cpp

```diff
@@ -42,12 +43,60 @@ namespace iroha {
           *pb_vote = PbConverters::serializeVote(vote);
         }
 
-        async_call_->Call([&](auto context, auto cq) {
-          return peers_.at(to.address())->AsyncSendState(context, request, cq);
-        });
-
         log_->info(
             "Send votes bundle[size={}] to {}", state.size(), to.address());
+
+        auto log_outcome = [log = log_, destination_peer = to.toString()](
+                               const grpc::Status &status) {
+          log->info("Sent to {} with status details [{}]",
+                    destination_peer,
+                    status.ok() ? "OK" : status.error_details());
+        };
+
+        return async_call_
+            ->Call([&](auto context, auto cq) {
+              return peers_.at(to.address())
+                  ->AsyncSendState(context, request, cq);
+            })
+            .tap(log_outcome)
+            .map(
+                [](const auto &status) { return makeSendStateStatus(status); });
+      }
+
+      YacNetworkWithFeedBack::ValueStateReturnType
+      NetworkImpl::makeSendStateStatus(const grpc::Status &status) {
+        auto is_ok = [](const auto &code) {
+          return code == grpc::StatusCode::OK;
+        };
+
+        auto is_troubles_with_recipient = [](const auto &code) {
+          using namespace grpc;
+          std::vector<int> codes = {StatusCode::CANCELLED,
+                                    StatusCode::INVALID_ARGUMENT,
+                                    StatusCode::UNAUTHENTICATED,
+                                    StatusCode::RESOURCE_EXHAUSTED,
+                                    StatusCode::ABORTED,
+                                    StatusCode::UNIMPLEMENTED,
+                                    StatusCode::UNAVAILABLE,
+                                    StatusCode::DATA_LOSS};
+          return std::any_of(codes.begin(), codes.end(), [code](auto val) {
```

Comment:
```suggestion
          return std::find(codes.begin(), codes.end(), code) != codes.end();
```
or, if using set,
```suggestion
          return codes.find(code) != codes.end();
```
which is faster